src/i_sdlsound.c

```diff
@@ -279,22 +273,73 @@ static void UnlockAllocatedSound(allocated_sound_t *snd)
     //printf("-- %s: Use count=%i\n", snd->sfxinfo->name, snd->use_count);
 }
 
+// Search through the list of allocated sounds and return the one that matches
+// the supplied sfxinfo entry and pitch level.
+
+static allocated_sound_t * GetAllocatedSoundBySfxInfoAndPitch(sfxinfo_t *sfxinfo, int pitch)
+{
+    allocated_sound_t * p = allocated_sounds_head;
+
+    while(p != NULL)
+    {
+        if(p->sfxinfo == sfxinfo && p->pitch == pitch)
+        {
+            return p;
+        }
+        p = p->next;
+    }
+    return NULL;
+}
+
+// Allocate a new sound chunk and pitch-shift an existing sound up-or-down
+// into it.
+
+static allocated_sound_t * PitchShift(allocated_sound_t *insnd, int pitch)
+{
+    allocated_sound_t * outsnd;
+    Sint16 *inp, *outp;
+    Sint16 *srcbuf, *dstbuf;
+    Uint32 srclen, dstlen;
+
+    dstlen = (insnd->chunk.alen)*((pitch+1)*100/NORM_PITCH)/100; // rounds up
```

Comment:
Which two? I guess one is #304. I don't disagree. It's probably the oldest line in the patch-set too. I might try a slightly different approach. It's also the one place at the moment to determine how much the pitch gets shifted, so I might need to calibrate it a bit based on more experiments with vanilla. (I've got some hacked EXEs and IWADs with a tuned sine wave of known pitch and length for all sfx and known RNG table to figure out the extent of the original shifting, I just need to do the recording and comparisons).